packages/runtime/container-runtime/src/blobManager/blobManager.ts

```diff
@@ -295,6 +305,17 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 		};
 	}
 
+	public async waitForStashedBlobs(): Promise<void> {
+		return Promise.all(this.pendingStashedBlobs.values())
```

Comment:
i think you should make this promise a singleton, rather than a new promise per call, and the singleton also clean itself up